meta-mender-qemu/scripts/mender-qemu

```diff
@@ -51,8 +51,17 @@ QEMU_ARGS=""
 # Extract certain variables from build, with "EXTRACTED_" prefix.
 eval "$(bitbake -e core-image-minimal | egrep '^(MACHINE|DISTRO_FEATURES)=' | sed -e 's/^/EXTRACTED_/')"
 
-# MACHINE is special, it can't be extracted from bitbake, so extract from local.conf.
-eval "$(sed -nre '/^MACHINE *\??\??= */{s/^MACHINE *\??\??= */EXTRACTED_MACHINE=/;p}' $BUILDDIR/conf/local.conf)"
+# Extract MACHINE definition from bitbake config files.
+# Process them in the order specified in bitbake.conf.
+# Note: it's possible this will miss some other possible configurations
+# but we don't want to have to run a full bitbake parse here.
+
+# Determine MACHINE if set with '='
+EXTRACTED_MACHINE="${EXTRACTED_MACHINE:-$(cat $BUILDDIR/conf/site.conf $BUILDDIR/conf/auto.conf $BUILDDIR/conf/local.conf 2>/dev/null | grep 'MACHINE *=' | grep -v '#' | tail -1 | cut -d= -f2 | tr -d \" | tr -d \ )}"
+# Determine MACHINE if set with '?='
+EXTRACTED_MACHINE="${EXTRACTED_MACHINE:-$(cat $BUILDDIR/conf/site.conf $BUILDDIR/conf/auto.conf $BUILDDIR/conf/local.conf 2>/dev/null | grep 'MACHINE *?=' | grep -v '#' | tail -1 | cut -d= -f2 | tr -d \" | tr -d \ )}"
+# Determine MACHINE if set with '??='
+EXTRACTED_MACHINE="${EXTRACTED_MACHINE:-$(cat $BUILDDIR/conf/site.conf $BUILDDIR/conf/auto.conf $BUILDDIR/conf/local.conf 2>/dev/null | grep 'MACHINE *??=' | grep -v '#' | tail -1 | cut -d= -f2 | tr -d \" | tr -d \ )}"
```

Comment:
> the only truly reliable way to figure it out is to run bitbake and let it parse the metadata and I don't think we want to do that on every invocation of this script.

Actually, we already do. Check out the `eval` above this section. I bet you can replace the `MACHINE` variable there with the new `MENDER_TEST_MACHINE` variable you added.

And following that, perhaps it should be renamed to `MENDER_MACHINE`, since it has uses outside of testing too.